### PR TITLE
New Observation Properties

### PR DESCRIPTION
**Pull request by swalker2m**
Adds two new observation properties and renames another.

* Add `visualizationTime`, which is used to compute the average parallactic position angle, proper-motion-adjusted coordinates and eventually non-sidereal target coordinates
* Add `posAngleConstraint` which is used to limit which values may be selected for the position angle
* Rename `config` to `manualConfig` to contrast it with the automatic sequence generation

The `PosAngleConstraint` differs from the core version because I found it difficult to map to the API.  What I ended up with seems simple to me, but I'm not sure and am open to suggestions.  To get a sense for it, skip down to the bottom where `MutationSuite` defines a few example mutations and results.

`PosAngleConstraint` is optional and can be deleted by setting it to `null`.  If not present, the the position angle is not constrained (i.e., `Unconstrained` in the core model).  Setting it to `AVERAGE_PARALLACTIC` and yet assigning a position angle results in a `ParallacticOverride` in the core model.  I think we can leave the core model alone and just write a codec to  do the translation.

### CODE REVIEW

**Comment by cquiroz** [comment]
Where would the code live? in `lucuma-schema`?

**Comment by swalker2m** [comment]
> Where would the code live? in `lucuma-schema`?

For the codec?  Good question.  If that makes sense I can write it and add it there.

File: modules/service/src/test/scala/MutationSuite.scala [hunk 1]
```diff
@@ -261,5 +261,170 @@ class MutationSuite extends OdbSuite {
...
+          }
+        }
+      }
+    """,
+    expected = json"""
+      {
+        "editObservation" : [
+          {
+            "id": "o-3",
+            "posAngleConstraint": null
```
Comment (cquiroz) [inline modules/service/src/test/scala/MutationSuite.scala:329]: This would be `Unconstrained` I guess

Comment (swalker2m) [reply modules/service/src/test/scala/MutationSuite.scala:329]: Exactly.

**Review by cquiroz — APPROVED** [review]
LGTM

**Pull request merged**